### PR TITLE
Feature updates

### PR DESCRIPTION
**Pull request by Aquaplant**
Overall: made more comfortable usage and preventing output console prints, if unwanted by user

Added:
 - new functions for better user-experience
 - parameters to existing functions to prevent warnings from PDB structure parser, fetching prints and made print of table of ignored residues optional

Updated:
- function documentation, changed some variable names for better comprehension, added parameter types
- outsourced some nested function for module-use
- wrapped resource stream (hopefully prevents IO warnings caused by unclosed stream)
- changed plot title / legend titles to file-name only
- README (to match actual changes)

Feedback is appreciated.

Edit: 
- TODO: Forgot to update module version
- used Python version 3.10

### CODE REVIEW

**Comment by Aquaplant** [comment]
Forgot to update module version

File: RamachanDraw/fetch.py [hunk 1]
```diff
@@ -1,12 +1,22 @@
-from Bio.PDB import PDBList
 from typing import Union
+from os.path import exists, realpath, basename
+from os import sep
```
Comment (alxdrcirilo) [inline RamachanDraw/fetch.py]: `sep` is never used, you can remove this line.

File: RamachanDraw/fetch.py [hunk 2]
```diff
@@ -1,8 +1,17 @@
 from typing import Union
-
+from os.path import exists, realpath, basename
```
Comment (alxdrcirilo) [inline RamachanDraw/fetch.py]: `basename` is never used, you can drop its import.

File: RamachanDraw/phi_psi.py [hunk 3]
```diff
@@ -1,51 +1,87 @@
-from Bio.PDB import PDBParser, PPBuilder
 from math import pi
+from typing import Union
+
+from Bio.PDB import PDBParser, PPBuilder
 from rich.console import Console
 from rich.table import Table
+from os.path import exists, realpath
```
Comment (alxdrcirilo) [inline RamachanDraw/phi_psi.py]: These are never used, you can remove this line.

File: RamachanDraw/phi_psi.py [hunk 4]
```diff
@@ -1,51 +1,87 @@
...
-    def get_ignored_res(file: str):
-        x, y, ignored, output = [], [], [], {}
-        for model in PDBParser().get_structure(id=None, file=file):
-            for chain in model:
-                peptides = PPBuilder().build_peptides(chain)
-                for peptide in peptides:
-                    for aa, angles in zip(peptide, peptide.get_phi_psi_list()):
-                        residue = chain.id + ":" + aa.resname + str(aa.id[1])
-                        output[residue] = angles
+def get_ignored_res(pdb_file_path: str, ignore_pdb_warnings: bool = False) -> (dict, list, list, list):
```
Comment (alxdrcirilo) [inline RamachanDraw/phi_psi.py]: Can you change the return type hint to `tuple[dict, list, list, list]` please?

File: RamachanDraw/plot.py [hunk 5]
```diff
@@ -1,43 +1,67 @@
...
-            if value[0] and value[1]:
-                x.append(value[0] * 180 / pi)
-                y.append(value[1] * 180 / pi)
-            else:
-                ignored.append((key, value))
 
-        return output, ignored, x, y
+def plot(pdb_file: Union[str, list, tuple], cmap: str = 'viridis', alpha: float = 0.75, dpi: int = 100,
+         save: bool = True, show: bool = False, out: str = 'plot.png',
+         ignore_pdb_warnings: bool = False) -> (plt.Axes, dict):
```
Comment (alxdrcirilo) [inline RamachanDraw/plot.py]: Can you change the return type hint to `tuple[plt.Axes, dict]` please?

File: RamachanDraw/plot.py [hunk 6]
```diff
@@ -1,43 +1,67 @@
...
+    path_out = []
+    for p in path:
+        path_split = p.split(os_separator)
+        pdb_name = path_split[-1]
+        if remove_extension:
+            pdb_name_split = pdb_name.split(".")
+            if len(pdb_name_split) > 1:
+                pdb_name = '.'.join(pdb_name_split[:-1])
+        path_out.append(pdb_name.upper() if upper_case else pdb_name)
+    return tuple(path_out)
```
Comment (alxdrcirilo) [inline RamachanDraw/plot.py:34]: This is a nice feature! :)

There may be a nicer way to do this, but for now it seems to get the job done.

**Review by alxdrcirilo — CHANGES_REQUESTED** [review]
Thanks for deciding to contribute!

Let's first merge this into the `dev` branch once the comments are resolved and the respective changes pushed.

I plan to dust off this project and modernize it a bit in the coming weeks. These changes will be included on top of the `dev` branch as well and merged together with your changes to `master` once everything is ready.

Good work! 👍

**Review by alxdrcirilo — APPROVED** [review]

**Pull request merged**